src/GraphQl/Subscription/SubscriptionManager.php

```diff
@@ -63,26 +73,129 @@ public function retrieveSubscriptionId(array $context, ?array $result, ?Operatio
+        if ($private && $privateFields && $previousObject) {
+            foreach ($options['private_fields'] as $privateField) {
+                unset($result['__private_field_'.$privateField]);
+            }
+        }
         $subscriptions[] = [$subscriptionId, $fields, $result];
         $subscriptionsCacheItem->set($subscriptions);
         $this->subscriptionsCache->save($subscriptionsCacheItem);
 
+        $this->updateSubscriptionCollectionCacheData(
+            $iri,
+            $fields,
+            $subscriptions,
+        );
+
         return $subscriptionId;
     }
 
-    public function getPushPayloads(object $object): array
+    public function getPushPayloads(object $object, string $type): array
+    {
+        if ('delete' === $type) {
+            $payloads =  $this->getDeletePushPayloads($object);
+        } else {
+            $payloads = $this->getCreatedOrUpdatedPayloads($object);
+        }
+
+        return $payloads;
+    }
+
+    /**
+     * @return array<array>
+     */
+    private function getSubscriptionsFromIri(string $iri): array
+    {
+        $subscriptionsCacheItem = $this->subscriptionsCache->getItem($this->encodeIriToCacheKey($iri));
+
+        if ($subscriptionsCacheItem->isHit()) {
+            return $subscriptionsCacheItem->get();
+        }
+
+        return [];
+    }
+
+    private function removeItemFromSubscriptionCache(string $iri): void
+    {
+        $cacheKey = $this->encodeIriToCacheKey($iri);
+        if ($this->subscriptionsCache->hasItem($cacheKey)) {
+            $this->subscriptionsCache->deleteItem($cacheKey);
+        }
+    }
+
+    private function encodeIriToCacheKey(string $iri): string
+    {
+        return str_replace('/', '_', $iri);
+    }
+
+    private function updateSubscriptionCollectionCacheData(
+        ?string                       $iri,
+        array                         $fields,
+        array                         $subscriptions,
+    ): void
+    {
+        $subscriptionCollectionCacheItem = $this->subscriptionsCache->getItem(
+            $this->encodeIriToCacheKey($this->getCollectionIri($iri)),
+        );
+        if ($subscriptionCollectionCacheItem->isHit()) {
+            $collectionSubscriptions = $subscriptionCollectionCacheItem->get();
+            foreach ($collectionSubscriptions as [$subscriptionId, $subscriptionFields, $subscriptionResult]) {
+                if ($subscriptionFields === $fields) {
+                    return;
+                }
+            }
+        }
+        $subscriptionCollectionCacheItem->set($subscriptions);
+        $this->subscriptionsCache->save($subscriptionCollectionCacheItem);
+    }
+
+    private function getResourceId(mixed $privateField, object $previousObject): string
+    {
+        $id = $previousObject->{'get' . ucfirst($privateField)}()->getId();
+        if ($id instanceof \Stringable) {
+            return (string)$id;
+        }
+        return $id;
+    }
```

Comment:
This is usually using `identifiersExtractor` could you explain why you need this?

Reply:
This is the part that extract the value from an object filed that is used for restrictive updates to be limited to specific context (in my case to a specific owner_id).

It also is extracting a defined field, not an ID, so the name for the method should be changed. 
I am all for doing it in a better way :)